packages/npm/@amazeelabs/react-framework-bridge/src/test-utils.ts

```diff
@@ -0,0 +1,9 @@
+import userEvent from '@testing-library/user-event';
+
+// After React 18 upgrade, `userEvent.type(..., 'bar')` produces wrong number of
+// events. So we type letter by letter.
```

Comment:
`userEvent.type(..., 'bar')` was producing
```
      expect(onChange).toHaveBeenCalledTimes(5);
      expect(onChange).toHaveBeenNthCalledWith(1, { foo: '' });
      expect(onChange).toHaveBeenNthCalledWith(2, { foo: 'b' });
      expect(onChange).toHaveBeenNthCalledWith(3, { foo: 'ba' });
      expect(onChange).toHaveBeenNthCalledWith(4, { foo: 'ba' });
      expect(onChange).toHaveBeenNthCalledWith(5, { foo: 'bar' });
```

This looks too shaky to me, so I created this little wrapper.